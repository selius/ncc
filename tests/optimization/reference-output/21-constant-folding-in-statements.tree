func:
{ }
`- return
   `- 7
main:
{ }
|- { }
|  |- =
|  |  |- a
|  |  `- 12
|  `- =
|     |- b
|     `- 10.88
|- pr()
|  |- a
|  `- b
|- if
|  |- 1
|  |- { }
|  |  `- __print_int()
|  |     `- 1
|  `- { }
|     `- __print_int()
|        `- 0
|- for
|  |- =
|  |  |- a
|  |  `- 4
|  |- <
|  |  |- a
|  |  `- 10
|  |- ++
|  |  `- a
|  `- { }
|     `- __print_int()
|        `- a
|- =
|  |- a
|  `- 10
|- while
|  |- >
|  |  |- a
|  |  `- 6
|  `- { }
|     `- __print_int()
|        `- --(postfix)
|           `- a
|- =
|  |- a
|  `- 3
|- do
|  |- { }
|  |  `- __print_int()
|  |     `- ++(postfix)
|  |        `- a
|  `- <
|     |- a
|     `- 7
|- __print_int()
|  `- func()
|- switch
|  |- 12
|  `- { }
|     |- case
|     |  |- 12
|     |  `- __print_int()
|     |     `- 1
|     |- break
|     |- case
|     |  |- 1
|     |  `- __print_int()
|     |     `- 0
|     |- break
|     |- case
|     |  |- 0
|     |  `- __print_int()
|     |     `- 0
|     `- break
`- return
   `- 0
pr:
{ }
|- __print_int()
|  `- a
`- __print_float()
   `- b
