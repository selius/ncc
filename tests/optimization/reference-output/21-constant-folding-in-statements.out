12
10.880000
1
4
5
6
7
8
9
10
9
8
7
3
4
5
6
7
1
